File: src/it/javacc-goal/src/main/csharp/Token.cs
/* CUSTOMIZED PARSER FILE - MUST NOT BE OVERWRITTEN BY JAVACC */
namespace BP7 {

public partial class Token {

  public int kind;

  public int beginLine;

  public int beginColumn;

  public int endLine;

  public int endColumn;

  public string image;

  public Token next;

  public Token specialToken;

  public object getValue() {
    return null;
  }

  public Token() {
  }

  public Token(int kind) : this(kind, null) {
  }

  public Token(int kind, string image) {
    this.kind = kind;
    this.image = image;
  }

  override public string ToString() {
    if (kind == 0) { // 0 is always EOF
      return "EOF";
    }
    return image;
  }

  public static Token newToken(int ofKind, string image) {
    switch(ofKind) {
      default :
        return new Token(ofKind, image);
    }
  }

  public static Token newToken(int ofKind) {
    return newToken(ofKind, null);
  }

} // end class

} // end namespace

File: src/it/javacc-goal/src/main/java/org/jcc/ßπ6/Token.java
/* CUSTOMIZED PARSER FILE - MUST NOT BE OVERWRITTEN BY JAVACC */
package org.jcc.ßπ6;

public class Token {

  public int kind;

  public int beginLine, beginColumn, endLine, endColumn;

  public String image;

  public Token next;

  public Token specialToken;

  public Token() {}

  public Token(final int kind) {
    this(kind, null);
  }

  public Token(final int kind, final String image) {
    this.kind = kind;
    this.image = image;
  }

  public Object getValue() {
    return null;
  }

  @Override
  public String toString() {
    return image;
  }

  public static Token newToken(final int ofKind, final String image) {
    switch (ofKind) {
      default:
        return new Token(ofKind, image);
    }
  }

  public static Token newToken(final int ofKind) {
    return newToken(ofKind, null);
  }
}
